### PR TITLE
fix: clean up documentation and workflows

### PR DESCRIPTION
**Pull request by hanno-jonlay**
Just opening a PR here so that we can track necessary changes to the documentation, as we refine our approach in the private repositories.

Remaining tasks:

- [x] Can the `yarn.lock` be removed?
- [x] Do we need to modify the documentation and the config to use the [`conventionalcommits`](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-conventionalcommits) preset? We want to ensure that a `build:` commit will trigger a new release, since this is quite common within the team. Our PR linter also uses this preset, so we will want the two to be consistent.
- [x] Add our PR Linting template to this repository (as per discussion on other projects)
- [x] The definition of our `inputs` in the README is quite hard to follow. Adopt the format [here](https://github.com/semantic-release/commit-analyzer) with tables, for clarity

### CODE REVIEW

File: flavours/release.js [hunk 1]
```diff
@@ -1,8 +1,22 @@
 module.exports = {
   branches: ['main'],
   plugins: [
-    '@semantic-release/commit-analyzer',
-    '@semantic-release/release-notes-generator',
+    [
```
Comment (hanno-jonlay) [inline flavours/release.js:4]: @hanno-marcelkalveram just in case you didn't have this on your list, these would need applying to `release-package` also, I think.

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (hanno-jonlay) [inline yarn.lock]: If we're making use of the lockfile like this, I presume we also need a `yarn install` step?

Could we also commit `@semantic-release/changelog` to the lockfile rather than manually installing it?

File: flavours/release.js (hunk 1 above)
Comment (hanno-marcelkalveram) [reply flavours/release.js:4]: I know! I wasn't done here yet 😄

File: yarn.lock (hunk 2 above)
Comment (hanno-marcelkalveram) [reply yarn.lock]: As mentioned above. I wasn't done here yet. I was just committing my changes before the break.

Comment (hanno-jonlay) [reply yarn.lock]: 😆 All good - I just got a few notifications and took a look 👍 

I won't look at this one again till you @ me when done!

**Comment by hanno-marcelkalveram** [comment]
> * [x]  Can the `yarn.lock` be removed?

I had to add the dependency `conventional-changelog-conventionalcommits` to use it inside the `config` field of the commit-analzer and release-notes plugins. That's why we now need the `yarn.lock` file for sure.

> * [x]  Do we need to modify the documentation and the config to use the [`conventionalcommits`](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-conventionalcommits) preset? We want to ensure that a `build:` commit will trigger a new release, since this is quite common within the team. Our PR linter also uses this preset, so we will want the two to be consistent.

I've tweaked the docs slightly and I saw that you already updated them.

> * [x]  Add our PR Linting template to this repository (as per discussion on other projects)

PR template is here: https://github.com/wearehanno/action-semantic-release/pull/9

File: action.yml [hunk 3]
```diff
@@ -15,11 +15,7 @@ runs:
   steps:
     - run: |
         yarn add \
-          @semantic-release/changelog \
-          @semantic-release/commit-analyzer \
-          @semantic-release/github \
-          @semantic-release/npm \
-          @semantic-release/release-notes-generator
+          @semantic-release/changelog
```
Comment (hanno-jonlay) [inline action.yml:18]: @hanno-marcelkalveram Don't we need to `yarn install` to make use of the dependencies you set up in the `yarn.lock`.

Is it also logical that we'd include this dependency in the lockfile too, so that we do not need to install a version like this on the fly, which is not necessarily locked at a specific version?

Comment (hanno-marcelkalveram) [reply action.yml]: I honestly haven't figured that out yet. AFAIK, `yarn install` would install the `package.json` dependencies of the repository it runs in. So I guess I'll have to install `conventional-changelog-conventionalcommits` manually.

I'll soon create a new private repository and try out this workflow to spot any issues.

Comment (hanno-jonlay) [reply action.yml]: That's fair.

I think it's worth trying out this PR on separate repository before we merge here, but I don't mind if we leave the `yarn` steps as-is for now. No need to over-engineer those at the moment.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -1,10 +1,10 @@
-name: Release
+name: "main"
 
 on: push
 
 jobs:
-  release:
-    name: release
+  main_release:
+    if: github.ref == 'refs/heads/main'
```
Comment (hanno-jonlay) [inline .github/workflows/main.yml:7]: @hanno-marcelkalveram FYI I just added this so that the action for releasing this project itself doesn't keep trying to run on branches.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -13,5 +13,5 @@ jobs:
       - name: Release
         uses: wearehanno/action-semantic-release@main
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Required for publishing a GitHub *Release*
           NPM_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (hanno-jonlay) [inline .github/workflows/main.yml:17]: Correct me if I'm wrong, but since we're only creating a release, not a package, we could also delete this line - right?

Should be no need to pass an `NPM_TOKEN` for releasing this repository.

**Review by hanno-jonlay — COMMENTED** [review]
Just 2 final comments ahead of the merge, I think.

Lmk if you disagree.

File: .github/workflows/main.yml (hunk 5 above)
Comment (hanno-marcelkalveram) [reply .github/workflows/main.yml:17]: This is still needed for the `release-package` workflow, isn't it? For the default one, this wouldn't be required, so it can be empty. But I don't know yet how to conditionally set env variables.

Comment (hanno-marcelkalveram) [reply .github/workflows/main.yml:17]: Oh boy. I'm getting confused all the time. This is the self-publishing workflow. Ok, so we don't need it 👍

File: .github/workflows/main.yml (hunk 4 above)
Comment (hanno-marcelkalveram) [reply .github/workflows/main.yml:7]: Ok 👍

File: action.yml (hunk 3 above)
Comment (hanno-marcelkalveram) [reply action.yml]: Ok, will do.

**Review by hanno-marcelkalveram — APPROVED** [review]

**Comment by hanno-jonlay** [comment]
Feel free to merge this one whenever you're ready.

**Comment by hanno-marcelkalveram** [comment]
Just testing this PR in other repos, to make sure it works as expected. Already spotted an issue. I'll let you know when it's all done.

File: .github/workflows/main.yml (hunk 5 above)
Comment (hanno-jonlay) [reply .github/workflows/main.yml:17]: OK, do you want to remove it, then? 😆 

I saw you resolved this conversation, but I think that task had to be completed.

Comment (hanno-marcelkalveram) [reply .github/workflows/main.yml]: Ah, true. I just committed that.

File: action.yml (hunk 3 above)
Comment (hanno-marcelkalveram) [reply action.yml]: This does work as expected now: https://github.com/wearehanno/marcel-testing/actions/runs/512076837.

Comment (hanno-marcelkalveram) [reply action.yml]: The only mistake I made is that I tried to publish it to a non-org, e.g. when the `name` in `package.json` is just `marcel-testing`, and not `@wearehanno/marcel-testing`, the release process fails because it can't figure out where to publish it.

**Comment by hanno-marcelkalveram** [comment]
@hanno-jonlay this is working now after one more fix: https://github.com/wearehanno/action-semantic-release/pull/8/commits/aba7d4dd4abe13ea9d113074275692f072953075

I've also changed the `build` type to release a `patch`, rather than `minor` release: https://github.com/wearehanno/action-semantic-release/pull/8/commits/bbcc4587e5894a0f8b7143afbf15750d5d8c038e. LMK if that's ok.

**Review by hanno-marcelkalveram — APPROVED** [review]
